app/controllers/report_controller/widgets.rb

```diff
@@ -655,7 +655,17 @@ def widget_set_record_vars(widget)
     #      end
     #      widget.miq_widget_shortcuts = ws
     #    end
-    widget.options[:url] = @edit[:new][:url] unless @edit[:new][:url].blank?
+    if @sb[:wtype] == "rf"
+      if @edit[:new][:feed_type] == "internal"
+        widget.resource = RssFeed.find(@edit[:new][:rss_feed_id]) if @edit[:new][:rss_feed_id]
+        widget.options.delete(:url)
+      else
+        widget.resource = nil
+        widget.options[:url] = @edit[:new][:url]
+      end
+    else
+      widget.resource = @edit[:rpt]
```

Comment:
:+1: To putting this in the MiqWidget model. Seems like there is a lot of model specific knowledge in this logic.